File: uikit/src/main/java/io/agora/chat/uikit/provider/EaseEmojiconInfoProvider.java
package io.agora.chat.uikit.provider;


import java.util.Map;

import io.agora.chat.uikit.models.EaseEmojicon;

public interface EaseEmojiconInfoProvider {

    /**
     * return EaseEmojicon for input emojiconIdentityCode
     * @param emojiconIdentityCode
     * @return
     */
    EaseEmojicon getEmojiconInfo(String emojiconIdentityCode);

    /**
     * get Emojicon map, key is the text of emoji, value is the resource id or local path of emoji icon(can't be URL on internet)
     * @return
     */
    Map<String, Object> getTextEmojiconMapping();
}

File: ease-im-kit/src/main/kotlin/io/agora/uikit/provider/EaseEmojiconInfoProvider.kt
package io.agora.uikit.provider

import io.agora.uikit.model.EaseEmojicon

/**
 * Custom emoji icon provider
 */
interface EaseEmojiconInfoProvider {
    /**
     * return EaseEmojicon for input emojiconIdentityCode
     * @param emojiconIdentityCode
     * @return
     */
    fun getEmojiconInfo(emojiconIdentityCode: String?): EaseEmojicon?

    /**
     * get Emojicon map, key is the text of emoji, value is the resource id or local path of emoji icon(can't be URL on internet)
     * @return
     */
    val textEmojiconMapping: Map<String?, Any?>?
}